ambry-store/src/main/java/com/github/ambry/store/StoreMetrics.java

```diff
@@ -259,6 +263,13 @@ void initializeIndexGauges(String storeId, final PersistentIndex index, final lo
           () -> index.getLogUsedCapacity() - blobStoreStats.getCachedValidSize().getSecond();
       registry.register(MetricRegistry.name(Log.class, prefix + "CurrentInvalidDataSize"), currentInvalidDataSize);
     }
+    if (enableStoreIndexDirectMemoryUsageMetric) {
+      indexes.putIfAbsent(storeId, index);
+      Gauge<Long> indexDirectMemoryUsage =
```

Comment:
Would this PersistentIndex::getDirectMemoryUsage take long time which will delay the emitting of metric?

Reply:
My guesstimation is that this might be slow, but it should be able to finish in the scope of 100s of milliseconds at worst. It depends on how many index segments we have on all the disks. 
I remember metric would be taken every minute, so it shouldn't be an issue. But just in case, I put a configuration here .

Reply:
nit: probably you can add some debug log to get an idea of how long it will be take in difference cases, up to you.

Reply:
done